### PR TITLE
update compiler options grammar

### PR DESCRIPTION
**Pull request by grianbrcom**

### CODE REVIEW

File: server/src/main/antlr4/org/eclipse/lsp/cobol/core/parser/CobolParser.g4 [hunk 1]
```diff
@@ -25,130 +25,137 @@ endProgramStatement
...
+   | AWO | NOAWO
+   | BLOCK0 | NOBLOCK0
+   | (BUFSIZE | BUF) LPARENCHAR (integerLiteral | INTEGERLITERAL_WITH_K) RPARENCHAR
+   | CICS (LPARENCHAR literal RPARENCHAR)? | NOCICS
+   | (CODEPAGE | CP) LPARENCHAR integerLiteral RPARENCHAR
+   | COMPILE | C_CHAR | (NOCOMPILE | NOC) (LPARENCHAR (S_CHAR | E_CHAR | W_CHAR) RPARENCHAR)?
+   | (COPYLOC | CPLC) LPARENCHAR ((SYSLIB | dataName) COMMACHAR)? (DSN | PATH) LPARENCHAR (literal | dataSetName) RPARENCHAR RPARENCHAR
+   | NOCOPYLOC | NOCPLC
+   | (COPYRIGHT | CPYR) LPARENCHAR literal RPARENCHAR | NOCOPYRIGHT | NOCPYR
+   | (CURRENCY | CURR) LPARENCHAR . RPARENCHAR | NOCURRENCY | NOCURR
```
Comment (temanbrcom) [inline server/src/main/antlr4/org/eclipse/lsp/cobol/core/parser/CobolParser.g4:48]: Does it allow anything inside the parenthesis?

Comment (grianbrcom) [reply server/src/main/antlr4/org/eclipse/lsp/cobol/core/parser/CobolParser.g4:48]: Yes, there can be any character like `€`, `$` or `₽`. And I have no idea how to do it the other way.

Comment (temanbrcom) [reply server/src/main/antlr4/org/eclipse/lsp/cobol/core/parser/CobolParser.g4:48]: Did you try this? https://stackoverflow.com/questions/14169820/regular-expression-to-match-all-currency-symbols/14169836

Comment (grianbrcom) [reply server/src/main/antlr4/org/eclipse/lsp/cobol/core/parser/CobolParser.g4]: Yes, you are right.

File: server/src/main/antlr4/org/eclipse/lsp/cobol/core/parser/CobolLexer.g4 [hunk 2]
```diff
@@ -771,23 +858,29 @@ XCTL : X C T L;
...
+ZLEN : Z L E N;
+ZON : Z O N;
+ZONECHECK : Z O N E C H E C K;
+ZONEDATA : Z O N E D A T A;
 ZWB : Z W B;
 
 UNDERSCORECHAR : '_';
 
+INTEGERLITERAL_WITH_K: INTEGERLITERAL ('K' | 'k');
+CURRENCY_SYMBOL : [\p{Sc}];
```
Comment (temanbrcom) [inline server/src/main/antlr4/org/eclipse/lsp/cobol/core/parser/CobolLexer.g4:882]: Please, add a test for the symbol and provide a replaceable token for it in `SpecialTokenHandling.properties`

**Review by temanbrcom — CHANGES_REQUESTED** [review]

File: server/src/main/antlr4/org/eclipse/lsp/cobol/core/parser/CobolLexer.g4 (hunk 2 above)
Comment (grianbrcom) [reply server/src/main/antlr4/org/eclipse/lsp/cobol/core/parser/CobolLexer.g4:882]: I've done those tests as a part of the UseCases set.
INTEGERLITERAL_WITH_K : https://github.com/grianbrcom/che-che4z-lsp-for-cobol/blob/compilerOptions/server/src/test/java/org/eclipse/lsp/cobol/usecases/TestProcessCbl.java#L77
CURRENCY_SYMBOL : https://github.com/grianbrcom/che-che4z-lsp-for-cobol/blob/compilerOptions/server/src/test/java/org/eclipse/lsp/cobol/usecases/TestProcessCbl.java#L108

Comment (temanbrcom) [reply server/src/main/antlr4/org/eclipse/lsp/cobol/core/parser/CobolLexer.g4:882]: But it contains only a positive case. What will happen if I put `CURR(foo)`?

**Review by temanbrcom — APPROVED** [review]
🥃

**Pull request merged**